lnd_test.go

```diff
@@ -99,12 +101,109 @@ func testBasicChannelFunding(net *networkHarness, t *testing.T) {
 	assertTxInBlock(block, closingTxid, t)
 }
 
+func testChannelBalance(net *networkHarness, t *testing.T) {
+	ctxb := context.Background()
+
+	openChannel := func(alice *lightningNode, bob *lightningNode, amount btcutil.Amount) *lnrpc.ChannelPoint {
+		chanOpenUpdate, err := net.OpenChannel(ctxb, alice, bob, amount, 1)
+		if err != nil {
+			t.Fatalf("unable to open channel: %v", err)
+		}
+
+		// Mine a block, then wait for Alice's node to notify us that the
+		// channel has been opened. The funding transaction should be found
+		// within the newly mined block.
+		blockHash, err := net.Miner.Node.Generate(1)
+		if err != nil {
+			t.Fatalf("unable to generate block: %v", err)
+		}
+		block, err := net.Miner.Node.GetBlock(blockHash[0])
+		if err != nil {
+			t.Fatalf("unable to get block: %v", err)
+		}
+		fundingChanPoint, err := net.WaitForChannelOpen(chanOpenUpdate)
+		if err != nil {
+			t.Fatalf("error while waiting for channeel open: %v", err)
+		}
+		fundingTxID, err := wire.NewShaHash(fundingChanPoint.FundingTxid)
+		if err != nil {
+			t.Fatalf("unable to create sha hash: %v", err)
+		}
+		assertTxInBlock(block, fundingTxID, t)
+
+		// The channel should be listed in the peer information returned by
+		// both peers.
+		chanPoint := wire.OutPoint{
+			Hash:  *fundingTxID,
+			Index: fundingChanPoint.OutputIndex,
+		}
+		err = net.AssertChannelExists(ctxb, alice, &chanPoint)
+		if err != nil {
+			t.Fatalf("unable to assert channel existence: %v", err)
+		}
+
+		return fundingChanPoint
+	}
+
+	closeChannel := func(node *lightningNode, fundingChanPoint *lnrpc.ChannelPoint) {
+		closeUpdates, err := net.CloseChannel(ctxb, node, fundingChanPoint, false)
+		if err != nil {
+			t.Fatalf("unable to clsoe channel: %v", err)
+		}
+
+		// Finally, generate a single block, wait for the final close status
+		// update, then ensure that the closing transaction was included in the
+		// block.
+		blockHash, err := net.Miner.Node.Generate(1)
+		if err != nil {
+			t.Fatalf("unable to generate block: %v", err)
+		}
+		block, err := net.Miner.Node.GetBlock(blockHash[0])
+		if err != nil {
+			t.Fatalf("unable to get block: %v", err)
+		}
+
+		closingTxid, err := net.WaitForChannelClose(closeUpdates)
+		if err != nil {
+			t.Fatalf("error while waiting for channel close: %v", err)
+		}
+		assertTxInBlock(block, closingTxid, t)
+
+	}
+
+	checkChannelBalance := func (node lnrpc.LightningClient, amount btcutil.Amount) {
+		response, err := node.ChannelBalance(ctxb, &lnrpc.ChannelBalanceRequest{})
+		balance := btcutil.Amount(response.Balance)
```

Comment:
The error (which is checked below) should be examined before accessing the `Balance` field. Otherwise, this may trigger an unnecessary panic in the case that `ChannelBalance` returns an error along with a nil response.